dash/dash.py

```diff
@@ -676,6 +679,19 @@ def _generate_meta_html(self):
 
     # Serve the JS bundles for each package
     def serve_component_suites(self, package_name, path_in_package_dist):
+        # Check if the resource has a fingerprint
+        res = re.match(
+            r'^(.*)[.]v\d+_\d+_\d+(-\w+[.]?\d+)?m\d{10}([.]js)$',
```

Comment:
Isn't this going to be used for `.css` and `.js.map` too? Anyway those are included in the `mimetype` dict below.

Also wondering if we really need to be that specific with the pattern, and if it really needs to have a `.` in the `(-\w+[.]?\d+)` group? What about:
```
r'^(.*)[.]v[\w_-]+m\d+([.][\w.]+)$'
```

Reply:
Good point about the various extensions.

`.js.map` is not really an issue here as the request resource comes from the `//# sourceMappingURL=bundle.js.map` at the end of the bundle file and will never get fingerprinted in the front-end or be provided by the sever in the html.

As for the regex being stronger vs. weaker, as I was coming at it from the perspective that we have full control over the format that will be provided, so might as well make a strong check here too. That said, there's no obvious advantage to making a stronger check and maybe we end up losing some flexibility if edge-cases appear. Will update with a more general pattern.

Reply:
OK cool, I'll buy that about the maps - if you tell me you've seen it working 😅 

Yeah my thinking about the regex was just it's safer to match anything we're sure *won't* be in the regular filename. Like, what if there's some system where mod time isn't 10 digits for whatever reason, or someone has a really nonstandard version identifier.

The `mimetype` dict as written right now is explicit - whether or not intentionally so - that the final extension can only be `js`, `css`, or `map` - anything else will throw a `KeyError`. Currently anything else you need would go through assets (which would be interesting to investigate for caching, though generally much less important than component js), not component suites.

Reply:
> if you tell me you've seen it working 😅

Yup. They work :) The "routing/fingerprint" done by Dash only affects the resources put in the HTML, in that sense it's more trickery than routing 😸 

For the regex, generalized the pattern to:
`^(?P<filename>.*)[.]v[\w-]+m[0-9a-fA-F]+(?P<extension>(?:(?:(?<![.])[.])?[\w@])+)$` and its inverse `^(?P<filename>[\w@-]+)(?P<extension>.*)$` which behaves as followed for various valid / invalid strings:

![image](https://user-images.githubusercontent.com/8092993/67497603-36de5a80-f64c-11e9-8c69-1d62fb2956e2.png)

![image](https://user-images.githubusercontent.com/8092993/67497631-3e056880-f64c-11e9-8a16-a41e4e526687.png)

These strings are now tested in unit tests within Dash.

> The mimetype dict as written right now is explicit

My understanding is that this was intentional. Learned about it when I added the `map` extension. There's apparently a library that offers mimetype mapping but I was told at the time that it was sketchy. Needs to be revisited. If only for the woff/ddk issue.